src/bootstrap/lib.rs

```diff
@@ -1034,13 +1034,17 @@ impl Build {
     }
 
     fn debuginfo_map_to(&self, which: GitRepo) -> Option<String> {
-        if !self.config.rust_remap_debuginfo {
+        if !self.config.rust_remap_debuginfo && self.config.download_rustc_commit.is_none() {
```

Comment:
why would this matter? if we're not remapping debuginfo, we shouldn't need this either way, right?

Reply:
Compiletest only replaces the `/rustc/$sha1` paths with a placeholder when the `CFG_VIRTUAL_RUST_SOURCE_BASE_DIR` environment variable is passed to it at build time, and right now this is the way to set that.

I guess I could change the implementation and instead of changing this setting globally, pass to compiletest another prefix to normalize when download-rustc is enable.d

Reply:
ah I see, it *needs* to remap the source directory because the one we've downloaded from CI is not in the path the tests expect. That seems reasonable, can you add it as a comment?

Reply:
> I guess I could change the implementation and instead of changing this setting globally, pass to compiletest another prefix to normalize when download-rustc is enabled.

Ended up with this approach, as there is way less risk of unintended side effects this way.